doc/examples/big_data_ingestion.py

```diff
@@ -0,0 +1,146 @@
+# where in each epoch it
+# shuffles and splits the result into ``num_shards``.
+
+###############################################################################
+# Feed the pipeline into trainers
+# -----------------------------------
+# Let’s also implement a ``TrainingWorker`` which consumes the shuffled data
+# from each shard.
+#
+# For simplicity, we will define a
+# `Ray Actor <https://docs.ray.io/en/latest/actors.html>`_ that emulates
+# training workers. Specifically,
+#
+# 1. It takes one shard of the shuffle pipeline for training;
+# 2. It iterates the shard to get per epoch training dataset;
+# 3. It then consumes the dataset by batches;
+
+
+@ray.remote
+class TrainingWorker:
+    def __init__(self, rank: int, shard: DatasetPipeline):
+        self.rank = rank
+        self.shard = shard
+
+    def train(self):
+        for epoch, training_dataset in enumerate(self.shard):
+            # Following code emulates epoch based SGD training.
+            # TODO: replace the code for real training.
+            for i, batch in enumerate(training_dataset.iter_batches()):
+                print(f"Training... epoch: {epoch}, batch: {i}")
+
+
+###########################################################################
+# Let's run it
+# -----------------------------
+# Now let’s run the data pipeline end-to-end:
+# We read parquet files from the example s3 bucket,
+# create the shuffle pipeline,
+# and use training workers to consume the pipeline.
+
+NUM_WORKERS = 4
+NUM_EPOCHS = 5
+
+TRAINING_DATA_100MB = "s3://ingest-100MB-example/"
+TRAINING_DATA_500GB = "s3://ingest-500GB-example/"
+
+ray.init()
+
+# TODO: use TRAINING_DATA_500GB when running on cluster.
+splits = create_shuffle_pipeline(TRAINING_DATA_100MB, NUM_WORKERS, NUM_EPOCHS)
+workers = [TrainingWorker(rank, shard) for rank, shard in enumerate(splits)]
+
+# Let's run the e2e pipeline
+ray.get([worker.train.remote() for worker in workers])
+
+#################################################################################
+# Scale the shuffle ingestion pipeline
+# --------------------------------------------------------
+#
+# Scale the shuffle ingestion pipeline is simple. With Ray, we can linearly
+# scale the pipeline from ingesting 100MB of data to 500GB of data simply by adding
+# more machines. To achieve that, we will run the same data pipeline we
+# just defined on a 74 nodes Ray cluster.
+#
+# First, we'll set up our Ray Cluster. The provided ``big_data_ingestion.yaml``
+# cluster config can be used to set up an AWS cluster with 70 CPU nodes and
+# 4 GPU nodes.
+#
+# **Step 1:** Bring up the Ray cluster.
+#
+# .. code-block:: bash
+#
+#     $ pip install ray boto3
+#     $ ray up big_data_ingestion.yaml
+#
+# **Step 2:**  Run ``big_data_ingestion.py`` on the cluster.
+#
+# .. code-block:: bash
+#
+#     $ ray rsync_up big_data_ingestion.yaml "./big_data_ingestion.py" \
+#       "~/big_data_ingestion.py"
+#     $ ray exec big_data_ingestion.yaml "~/big_data_ingestion.py"
```

Comment:
Does [`ray submit`](https://docs.ray.io/en/master/package-ref.html#ray-submit) not work here?
```suggestion
#     $ ray submit big_data_ingestion.yaml ./big_data_ingestion.py
```